confgenerator/files.go

```diff
@@ -24,17 +24,30 @@ import (
 	"github.com/shirou/gopsutil/host"
 )
 
-func ReadUnifiedConfigFromFile(path, platform string) (UnifiedConfig, error) {
-	uc := UnifiedConfig{}
+// ReadUnifiedConfigFromFile reads the user config file and returns a UnifiedConfig.
+// If the user config file does not exist, it returns nil.
+func ReadUnifiedConfigFromFile(path, platform string) (*UnifiedConfig, error) {
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			// If the user config file does not exist, we don't want any overrides.
+			return nil, nil
+		}
+		return nil, fmt.Errorf("failed to retrieve the user config file %q: %w \n", path, err)
+	}
 
+	uc := &UnifiedConfig{}
 	data, err := ioutil.ReadFile(path)
 	if err != nil {
 		return uc, err
 	}
-	uc, err = UnmarshalYamlToUnifiedConfig(data, platform)
+	*uc, err = UnmarshalYamlToUnifiedConfig(data, platform)
```

Comment:
UnmarshalYamlToUnifiedConfig really should return a reference, should we do that in this PR?

Reply:
Might as well. Done.